### PR TITLE
feat: allow listing file contents with `debug:log`

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

This further improves the debug functionality to be able to actually see what is the content of the files generated by the template

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.examples.ts [hunk 1]
```diff
@@ -48,4 +48,20 @@ export const examples: TemplateExample[] = [
...
+  {
+    description: 'List the workspace directory with file contents',
+    example: yaml.stringify({
+      steps: [
+        {
+          action: 'debug:log',
+          id: 'write-workspace-directory',
+          name: 'List the workspace directory with file contents',
+          input: {
+            listWorkspace: true,
```
Comment (Rugvip) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.examples.ts:60]: Potential option here since there's currently a hidden dependency on `listWorkspace` from `listFileContents`: Might make sense to have this be `listWorkspace?: 'with-contents' | boolean` or something like that instead

**Review by acierto — APPROVED** [review]

**Review by mounicarajput — APPROVED** [review]

**Review by freben — APPROVED** [review]

**Review by camilaibs — APPROVED** [review]
LGTM

**Review by benjdlambert — APPROVED** [review]
Nice

**Comment by alexef** [comment]
hey team, this PR removed the `extra` parameter from the input schema, which broke our automation templates.

Should the extra parameter make a comeback, please let us know.

If not, please mention it in the release notes 🙏

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.ts [hunk 2]
```diff
@@ -32,28 +34,24 @@ const id = 'debug:log';
...
-        properties: {
-          message: {
-            title: 'Message to output.',
-            type: 'string',
-          },
-          listWorkspace: {
-            title: 'List all files in the workspace, if true.',
-            type: 'boolean',
-          },
-          extra: {
```
Comment (alexef) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.ts:52]: why was this removed?

**Comment by gtfortytwo** [comment]
The unexpected removal of the `extra` parameter mentioned by @alexef also broke things for us. We'd love some feedback on why this was removed.

@alexef Have you pursued raising this as issue already?  It's been 4 months, seems nothing is likely to come of comments on this merged PR at this point.

**Comment by alexef** [comment]
@gtfortytwo no, I did not (afaik). however, I found a different way to log objects, i.e. 

```
    - id: debug
      name: Result
      action: debug:log
      input:
        message: '${{ steps["xxx"].output.body | dump }}'
```

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.ts (hunk 2 above)
Comment (alexef) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/debug/log.ts:52]: ping @drodil / @benjdlambert
